src/middlewared/middlewared/plugins/webdav.py

```diff
@@ -244,13 +253,19 @@ async def validate(self, data, schema_name):
                     'WebDAV SSL protocol specified without choosing a certificate'
                 )
             else:
+                cert_choices = await self.cert_choices()
+                cert_in_choices = any(filter(lambda x: x['id'] == cert_ssl, cert_choices))
+                if not cert_in_choices:
+                    verrors.add(
+                        f'{schema_name}.certssl',
+                        f'Certificate is not permitted. Valid choices are {[x["name"] for x in cert_choices]}'
```

Comment:
It's better to do `", ".join(x["name"] for x in cert_choices)`

Reply:
Btw is this even required ? The method below handles the validation already ?

Reply:
I am not sure why using the same cert as the UI one is a problem 🤔 ?

Reply:
It's a matter of web browser behavior. Firefox would not allow me to access our webui after accessing the webdav server first.

Reply:
Hmm... and now I'm not seeing the SSL error anymore. I can remove that part of validation then.

Reply:
Can you share the error you got when this happened please ? I think it shouldn't be an issue

Reply:
SEC_ERROR_REUSED_ISSUER_AND_SERIAL

Reply:
Okay. That was just an issue with a janky default cert on my server.